iroh-p2p/src/service.rs

```diff
@@ -160,21 +162,30 @@ impl Libp2pService {
                 // rpc_client.call("storage", "get", cid)
                 trace!("Don't have data for: {}", cid);
             }
-            Event::Kademlia(KademliaEvent::OutboundQueryCompleted { id, result, .. }) => {
+            Event::Kademlia(KademliaEvent::OutboundQueryCompleted { result, .. }) => {
                 info!("kad: {:?}", result);
                 match result {
-                    QueryResult::GetProviders(Ok(GetProvidersOk { providers, .. })) => {
-                        if let Some(QueryChannel::GetProviders(ch)) = self.kad_queries.remove(&id) {
-                            ch.send(Ok(providers)).ok();
+                    QueryResult::GetProviders(Ok(GetProvidersOk { providers, key, .. })) => {
+                        if let Some(QueryChannel::GetProviders(chans)) =
+                            self.kad_queries.remove(&key)
+                        {
+                            for chan in chans.into_iter() {
+                                debug!("Sending providers for {:?}", key);
+                                chan.send(Ok(providers.clone())).ok();
+                            }
```

Comment:
lets log sth if we get results that have no listeners, can totally happen, but good to know still